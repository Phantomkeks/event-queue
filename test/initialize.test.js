"use strict";

const path = require("path");

const eventQueue = require("../src");

describe("initialize", () => {
  test("read yaml config file", async () => {
    const configFilePath = path.join(__dirname, "asset", "configFaulty.yml");
    await eventQueue.initialize({ configFilePath, registerDbHandler: false });
    const config = eventQueue.getConfigInstance().events;
    expect(config).toMatchSnapshot();
  });

  test("read yaml config file", async () => {
    const configFilePath = path.join(__dirname, "asset", "config.kk");
    await expect(
      eventQueue.initialize({ configFilePath, registerDbHandler: false })
    ).rejects.toThrow();
  });
});
