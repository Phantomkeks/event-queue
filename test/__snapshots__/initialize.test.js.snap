// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`initialize read yaml config file 1`] = `
[
  {
    "impl": "./srv/notification/EventQueueNotificationProcessor",
    "load": 40,
    "parallelEventProcessing": 5,
    "subType": "Task",
    "type": "Notifications",
  },
  {
    "impl": "./srv/businessLogs/EventQueueBusinessLogProcessor",
    "load": 10,
    "parallelEventProcessing": 1,
    "subType": "CommunicationSystem",
    "type": "BusinessLogs",
  },
]
`;
